core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.js

```diff
@@ -144,6 +182,25 @@ oppia.factory('ExplorationStatesService', [
         var newStateData = _states.getState(stateName);
         var accessorList = PROPERTY_REF_DATA[backendName];
 
+        if (CONTENT_ID_EXTRACTION.hasOwnProperty(backendName)) {
+          var oldContentIds = CONTENT_ID_EXTRACTION[backendName](oldValue);
+          var newContentIds = CONTENT_ID_EXTRACTION[backendName](newValue);
+          var contentIdsToDelete = _getSetDifference(
+            oldContentIds, newContentIds);
+          var contentIdsToAdd = _getSetDifference(newContentIds, oldContentIds);
+          contentIdsToDelete.forEach(function(contentId) {
+            newStateData.contentIdsToAudioTranslations.deleteContentId(
```

Comment:
Just wondering -- why are there two items here that need to be deleted sequentially? Are these two things representing the same source of truth? If so, they should be one item...

Reply:
newStateData.contentIdsToAudioTranslations is from the object having entire exploration states, and StateContentIdsToAudioTranslations is from the current active state which have displayed and memento properties. We have to save this in both of the places I didn't find a proper place for this one I was thinking to use $broadcast, will that be fine or we can leave it over here?.

Reply:
Hm, that seems a bit odd, right? The newStateData is supposed to be the stable source-of-truth for the data of the object, but the .displayed field in StateContentIdsToAudioTranslations is supposed to be transient (the source-of-truth is located in savedMemento). Or am I misunderstanding?

Also, separately, do we actually need a separate StateContentIdsToAudioTranslations property at all -- is it possible/sensible to get rid of it altogether and just use newStateData as the source of truth?

In general I'm just wondering why this seems to be treated differently than the other property updates which just modify newStateData, and whether there is any way to align them more with that process.

Reply:
We do need StateContentIdsToAudioTranslations for translation and also in the editor page so that editor can mark "needs update" to existing audio.

>In general I'm just wondering why this seems to be treated differently than the other property updates which just modify newStateData, and whether there is any way to align them more with that process.
We can make similar functionality in StatePropertyServices.js and do a similar check.

Reply:
OK, let's do that? We shouldn't have discrepancies, it makes the code harder to understand.

Reply:
@seanlip, I realized that we aren't displaying any part of  contentIdsToAudioTranslations in editor tab (except "marking needs update" which also isn't visible.) so I initiated stateConetntIdsToAudioTranslationsService in the translation tab only. Done! PTAL@!